emcc.py

```diff
@@ -2225,12 +2227,12 @@ def get_final():
 
       shared.JS.handle_license(final)
 
-      if final_suffix in JS_ENDINGS:
-        js_target = target
-      elif final_suffix in WASM_ENDINGS:
+      if final_suffix in WASM_ENDINGS:
         js_target = misc_temp_files.get(suffix='.js').name
-      else:
+      elif final_suffix == '.html':
         js_target = unsuffixed(target) + '.js'
+      else:
+        js_target = target
```

Comment:
how is SIDE_MODULE handled by this if-else chain? It seems like the final suffix would be `.so` if  the user asked for that, so we'd reach the final `else`, and then `js_target = foo.so`? I must be misunderstanding how this works, sorry...

Reply:
We don't this far with SIDE_MODULE (see return statement just above) since there is no js generated for that case.

Reply:
Ah, that makes sense then. But then this is unrelated to the PR, isn't it? (if the PR is for SIDE_MODULE, and this code is not relevant for SIDE_MODULE - am I missing something?)

Reply:
Not sure about the temp directory specifically, but standalone mode still has a JS file, which can be used to load and run the wasm on the web (while the wasm can still be used standalone off the web).